cypress/integration/page.spec.js

```diff
@@ -1,7 +1,290 @@
+  sortByLengthButton: () => cy.contains('button', 'Sort by length'),
+  reverseButton: () => cy.contains('button', 'Reverse'),
+  resetButton: () => cy.contains('button', 'Reset'),
+  goods: () => cy.byDataCy('Good'),
+
+  assertFirstGoods: (first, second, third) => {
+    page.goods().eq(0).should('have.text', first);
+    page.goods().eq(1).should('have.text', second);
+    page.goods().eq(2).should('have.text', third);
+  }
+};
+
+const NOT_ACTIVE_CLASS = 'is-light';
+
+let failed = false;
+
+Cypress.on('fail', (e) => {
+  failed = true;
+  throw e;
+});
 
 describe('Page', () => {
-  it('should be visitable', () => {
+  beforeEach(() => {
+    if (failed) Cypress.runner.stop();
+
     cy.visit('/');
   });
+
+  describe('by default', () => {
+    it('should show all the goods in the initial order', () => {
+      page.goods().should('have.length', 10);
+      page.assertFirstGoods('Dumplings', 'Carrot', 'Eggs');
+    });
+
+    it('should not show "Reset" button', () => {
+      page.resetButton().should('not.exist');
+    });
+
+    it('should have not active "Sort alphabetically" button', () => {
+      page.sortAlphButton().should('have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should have not active "Sort by length" button', () => {
+      page.sortByLengthButton().should('have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should have not active "Reverse" button', () => {
+      page.reverseButton().should('have.class', NOT_ACTIVE_CLASS);
+    });
+  });
+
+  describe('after pressing "Reverse"', () => {
+    beforeEach(() => {
+      page.reverseButton().click();
+    });
+
+    it(`should reverse goods`, () => {
+      page.assertFirstGoods('Garlic', 'Jam', 'Honey');
+    });
+
+    it('should show "Reset" button', () => {
+      page.resetButton().should('exist');
+    });
+
+    it('should have not active "Sort alphabetically" button', () => {
+      page.sortAlphButton().should('have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should have not active "Sort by length" button', () => {
+      page.sortByLengthButton().should('have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should have active "Reverse" button', () => {
+      page.reverseButton().should('not.have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should allow to sort alphabetically', () => {
+      page.sortAlphButton().click();
+
+      page.assertFirstGoods('Jam', 'Ice cream', 'Honey');
+      page.reverseButton().should('not.have.class', NOT_ACTIVE_CLASS);
+      page.sortAlphButton().should('not.have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should allow to sort by length', () => {
+      page.sortByLengthButton().click();
+
+      page.assertFirstGoods('Ice cream', 'Dumplings', 'Garlic');
+      page.reverseButton().should('not.have.class', NOT_ACTIVE_CLASS);
+      page.sortByLengthButton().should('not.have.class', NOT_ACTIVE_CLASS);
+    });
+
+    it('should allow to reset', () => {
+      page.resetButton().click()
```

Comment:
Missing semicolon at the end of the line. Consider adding it to maintain consistency and avoid potential issues.